core/shared/src/main/scala/zio/ZSchedule.scala

```diff
@@ -339,19 +345,30 @@ trait ZSchedule[-R, -A, +B] extends Serializable { self =>
     new ZSchedule[R1, A1, B] {
       type State = self.State
       val initial = f(self.initial)
+      val extract = self.extract
       val update  = self.update
     }
 
   /**
-   * Applies random jitter to the schedule bounded by the factors 0.0 and 1.0.
+   * Applies random jitter to all sleeps executed by the schedule.
    */
-  final def jittered: ZSchedule[R with Random, A, B] = jittered(0.0, 1.0)
+  final def jittered(min: Double = 0.0, max: Double = 1.0)(implicit ev: ZEnv <:< R): ZSchedule[ZEnv, A, B] =
```

Comment:
Why not just increase the delay by a random amount? My understanding is that the main use case of `jittered` is to make sure that that we are not inadvertently having a bunch of effects occur at the exact same time and, for example, overwhelming a server. This could be achieved by just adding an additional random delay versus modifying the delay. This would provide a nicer API because we would not have to get into any of these issues about mapping the environment or needing the environment to be a particular type. We could always provide another method that can potentially decrease the delay for users who need that.

I think we should be able to get to the point where for common use cases the environment type can be fully generic and doesn't require special modification and then modifying the environment can be a feature for power users to achieve functionality not available through a fully generic environment type.

Reply:
That's probably a good point for `jittered` in particular. Of course we still need things like `modifyDelay`, etc.

Reply:
Yes exactly.

Reply:
I have added ZSchedule#randomDelay and ZSchedule#randomDelayNormal that create a schedule that sleeps with a uniformly and normally distributed delay respectively. Using >>> this can be used to add random delays to arbitrary schedules.

I think jittered is also worth keeping around the way it currently exists.